pkg/operator/webhooksupportabilitycontroller/degraded_webhook.go

```diff
@@ -0,0 +1,104 @@
+package webhooksupportabilitycontroller
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"sort"
+	"strings"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// webhookInfo generically represents a webhook
+type webhookInfo struct {
+	Name     string
+	Service  *serviceReference
+	CABundle []byte
+}
+
+// serviceReference generically represents a service reference
+type serviceReference struct {
+	Namespace string
+	Name      string
+	Port      *int32
+}
+
+// updateWebhookConfigurationDegraded updates the condition specified after
+// checking that the services associated with the specified webhooks exist
+// and have at least one ready endpoint.
+func (c *webhookSupportabilityController) updateWebhookConfigurationDegraded(ctx context.Context, condition operatorv1.OperatorCondition, webhookInfos []webhookInfo) v1helpers.UpdateStatusFunc {
+	var serviceMsgs []string
+	var tlsMsgs []string
+	for _, webhook := range webhookInfos {
+		if webhook.Service != nil {
+			err := c.assertService(webhook.Service)
+			if err != nil {
+				serviceMsgs = append(serviceMsgs, fmt.Sprintf("%s: %s", webhook.Name, err))
+				continue
+			}
+			err = c.assertConnect(ctx, webhook.Service, webhook.CABundle)
+			if err != nil {
+				tlsMsgs = append(tlsMsgs, fmt.Sprintf("%s: %s", webhook.Name, err))
+				continue
+			}
+		}
+	}
+
+	svc, tls := len(serviceMsgs) > 0, len(tlsMsgs) > 0
+	switch {
+	case svc && tls:
+		condition.Reason = WebhookServiceNotReadyReason
+		condition.Status = operatorv1.ConditionTrue
+	case svc:
+		condition.Reason = WebhookServiceNotFoundReason
+		condition.Status = operatorv1.ConditionTrue
+	case tls:
+		condition.Reason = WebhookServiceConnectionErrorReason
+		condition.Status = operatorv1.ConditionTrue
+	default:
+		condition.Reason = ""
+		condition.Status = operatorv1.ConditionFalse
+	}
+	msgs := append(serviceMsgs, tlsMsgs...)
+	sort.Strings(msgs)
+	condition.Message = strings.Join(msgs, "\n")
+
+	return v1helpers.UpdateConditionFn(condition)
+}
+
+// assertService checks that the referenced service resource exists.
+func (c *webhookSupportabilityController) assertService(reference *serviceReference) error {
+	_, err := c.serviceLister.Services(reference.Namespace).Get(reference.Name)
```

Comment:
How long is the inertia of these checks regarding the operator degraded condition? Wondering whether we run into false positives e.g. when CVO or OLM install a new CRD. An inertia of 1 or 2 min would be good IMO.

Reply:
The status controller [defaults to a 2 minute inertia on degraded status](https://github.com/openshift/library-go/blob/master/pkg/operator/status/status_controller.go#L75).